pkg/internal/cyberark/identity/identity.go

```diff
@@ -174,15 +174,15 @@ type advanceAuthenticationResponseResult struct {
 }
 
 // Client is an client for interacting with the CyberArk Identity API and performing a login using a username and password.
-// For context on the behaviour of this client, see the Pytho SDK: https://github.com/cyberark/ark-sdk-python/blob/3be12c3f2d3a2d0407025028943e584b6edc5996/ark_sdk_python/auth/identity/ark_identity.py
+// For context on the behaviour of this client, see the Python SDK: https://github.com/cyberark/ark-sdk-python/blob/3be12c3f2d3a2d0407025028943e584b6edc5996/ark_sdk_python/auth/identity/ark_identity.py
 type Client struct {
 	client *http.Client
 
 	endpoint  string
 	subdomain string
 
-	tokenCache      map[string]token
-	tokenCacheMutex sync.Mutex
```

Comment:
suggestion: This change means that if the auth credentials change we'll continue to send the token for the old username, right?

It seems reasonable for us to handle a change of credentials while the agent is running; this would require that the username is passed in when the client is doing an upload, but that seems doable, right?

Was there a different motivation for this change?

Reply:
> It was not clear to me in what situation the auth credentials would change.

If they're provided as a secret which we read at runtime. It doesn't really matter I guess.